extension/encoding/googlecloudlogentryencodingextension/internal/proxynlb/parser.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package proxynlb contains utilities for parsing Google Cloud Proxy Network Load Balancer logs.
+package proxynlb // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/googlecloudlogentryencodingextension/internal/proxynlb"
+
+import (
+	"fmt"
+	"time"
+
+	gojson "github.com/goccy/go-json"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	semconv "go.opentelemetry.io/otel/semconv/v1.34.0"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/googlecloudlogentryencodingextension/internal/shared"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/googlecloudlogentryencodingextension/internal/vpcflowlog"
+)
+
+const (
+	// ConnectionsLogNameSuffix identifies load balancer connection logs in the logName field.
+	ConnectionsLogNameSuffix = "loadbalancing.googleapis.com%2Fconnections"
+
+	loadBalancerLogType = "type.googleapis.com/google.cloud.loadbalancing.type.LoadBalancerLogEntry"
+
+	gcpProxyNLBConnectionStartTime = "gcp.load_balancing.proxy_nlb.connection.start_time"
+	gcpProxyNLBConnectionEndTime   = "gcp.load_balancing.proxy_nlb.connection.end_time"
+	gcpProxyNLBServerBytesReceived = "gcp.load_balancing.proxy_nlb.server.bytes_received"
+	gcpProxyNLBServerBytesSent     = "gcp.load_balancing.proxy_nlb.server.bytes_sent"
+)
+
+type loadBalancerLog struct {
+	Type                string      `json:"@type"`
+	Connection          *connection `json:"connection"`
+	StartTime           *time.Time  `json:"startTime"`
+	EndTime             *time.Time  `json:"endTime"`
+	ServerBytesReceived string      `json:"serverBytesReceived"`
+	ServerBytesSent     string      `json:"serverBytesSent"`
+}
+
+type connection struct {
+	ClientIP   string `json:"clientIp"`
+	ClientPort *int64 `json:"clientPort"`
+	Protocol   *int64 `json:"protocol"`
+	ServerIP   string `json:"serverIp"`
+	ServerPort *int64 `json:"serverPort"`
+}
+
+// ParsePayloadIntoAttributes unmarshals the provided payload into the supplied attribute map.
+func ParsePayloadIntoAttributes(payload []byte, attr pcommon.Map) error {
+	var log loadBalancerLog
+	if err := gojson.Unmarshal(payload, &log); err != nil {
+		return fmt.Errorf("failed to unmarshal Proxy NLB log payload: %w", err)
+	}
+
+	if log.Type != "" && log.Type != loadBalancerLogType {
+		return fmt.Errorf("unexpected log type %q, expected %q", log.Type, loadBalancerLogType)
+	}
+
+	handleConnection(log.Connection, attr)
+	handleTimestamps(log.StartTime, log.EndTime, attr)
+
+	if err := shared.AddStrAsInt(gcpProxyNLBServerBytesReceived, log.ServerBytesReceived, attr); err != nil {
```

Comment:
So, this is kinda weird, but works. In [google's docs](https://docs.cloud.google.com/load-balancing/docs/tcp/tcp-ssl-proxy-logging-monitoring) they say this would be an integer... but it's actually sent as a string encoded integer. Perhaps to avoid JS float issues? Perhaps it's a bug? I'm not sure why. You can see it in the testdata `.json` files later.